File: services/auth/src/main/java/com/example/iter/auth/dto/response/AddressResponse.java
package com.example.iter.auth.dto.response;

import com.example.iter.auth.domain.entity.UserAddress;

public record AddressResponse(
        Long id,
        String recipientName,
        String recipientPhone,
        String zipcode,
        String address,
        String detailAddress
) {
    public static AddressResponse from(UserAddress address) {
        return new AddressResponse(
                address.getId(),
                address.getRecipientName(),
                address.getRecipientPhone(),
                address.getZipcode(),
                address.getAddress(),
                address.getDetailAddress()
        );
    }
}

File: services/auth/src/main/kotlin/iter/auth/dto/response/AddressResponse.kt
package iter.auth.dto.response

import iter.auth.domain.entity.UserAddress

@JvmRecord
data class AddressResponse(
    val id: Long?,
    val recipientName: String,
    val recipientPhone: String,
    val zipcode: String,
    val address: String,
    val detailAddress: String,
) {
    companion object {
        @JvmStatic
        fun from(address: UserAddress): AddressResponse =
            AddressResponse(
                address.id,
                address.recipientName,
                address.recipientPhone,
                address.zipcode,
                address.address,
                address.detailAddress,
            )
    }
}
